spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -885,8 +886,7 @@ protected MessageHandler getErrorMessageHandler(final ConsumerDestination destin
 
 			return (message) -> {
 
-				final ConsumerRecord<Object, Object> record = message.getHeaders()
-						.get(KafkaHeaders.RAW_DATA, ConsumerRecord.class);
+				ConsumerRecord<Object, Object> record = message.getHeaders().get(IntegrationMessageHeaderAccessor.SOURCE_DATA, ConsumerRecord.class);
```

Comment:
`ConsumerRecord<Object, Object> record = StaticMessageHeaderAccessor.getSourceData(message);`

Is a little simpler.